Modular saltworks

Maleclypse opened the pull request:
#### Summary
Summary: Content "Faction Camp Modular Saltworks"

#### Purpose of change
Creating an additional modular camp expansion.  Modeled after Curstwist's PRs in the same topic.  

#### Describe the solution
Creating additional json files similar the livestock and storage campsite PR's.  

#### Describe alternatives you've considered
Continuing to just make single json edits. 

#### Additional context
"WIP" I've managed to get the game to load without load stopping errors but I can't get the expand base mission to complete successfully and create the modular campsite Saltworks. Any suggestions very welcome.  
Once I can get one version to create I plan to create other building material versions and work on the recipe mission list. 

I'm pretty ready to close this out but I think I'm going to need some help to finish it in a reasonable timeline.

Maleclypse committed:
Merge pull request #1 from CleverRaven/master

Update

Maleclypse committed:
Merge pull request #2 from CleverRaven/master

Catching up to current version

Maleclypse committed:
WIP Saltworks

Creating Modular basecamp Saltworks with a Brewery. Still needs recipes for follower missions.  Still needs additional building materials versions but trying to get the camp site to build.

Maleclypse commented:
![Error recipe group](https://user-images.githubusercontent.com/54345792/65366599-09893180-dbf4-11e9-85a5-9e36de48b3c8.png)
![failed to add saltworks](https://user-images.githubusercontent.com/54345792/65366600-0e4de580-dbf4-11e9-95a5-369d85f3d0ef.png)

ZhilkinSerg commented on data/json/recipes/basecamps/recipe_modular_saltworks_log.json:
```diff
@@ -0,0 +1,77 @@
+    "blueprint_excludes": [ { "id": "fbms_souththwest" } ],
+    "blueprint_autocalc": true
+  }
+]
```
```suggestion
]

```

Maleclypse committed:
Update data/json/recipes/basecamps/recipe_modular_saltworks_log.json

Co-Authored-By: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

ZhilkinSerg commented on data/json/mapgen/basecamps/modular_saltworks_log.json:
```diff
@@ -0,0 +1,141 @@
+    "method": "json",
+    "object": { "place_nested": [ { "chunks": [ "fbml_stable2_log" ], "x": 3, "y": 14 } ] }
+  }
+]
```
```suggestion
]

```

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update data/json/mapgen/basecamps/modular_saltworks_log.json

Co-Authored-By: ZhilkinSerg <ZhilkinSerg@users.noreply.github.com>

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Trying to fix this

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update modular_saltworks_log.json

Maleclypse commented:
@curstwist I hate to bother you, but if you have a second to look over this at any time and if you see where I'm going wrong I'd appreciate the help.  If no time, no worries I'll muddle through :)

curstwist commented:
Looks like you are missing the survey recipe step.

Maleclypse commented:
> Looks like you are missing the survey recipe step.
> 
> And its mapgen partner. Look at the 1st entries in a _wood recipe &mapgen file. I dont repeat it in other palettes
> 
> ```
> {
>         "type": "recipe",
>         "result": "faction_base_workshop_0",
>         "description": "Survey land for a workshop area.",
>         "category": "CC_BUILDING",
>         "subcategory": "CSC_BUILDING_BASES",
>         "skill_used": "fabrication",
>         "autolearn": false,
>         "never_learn": true,
>         "construction_blueprint": "faction_base_workshop_0",
>         "blueprint_name": "workshop survey",
>         "time": "180 m",
>         "blueprint_requires": [ { "id": "not_an_upgrade" } ],
>         "blueprint_provides": [ { "id": "fbmw_0" } ]
>       },
> ```
> 
> and
> 
> ```
> {
>     "type": "mapgen",
>     "method": "json",
>     "nested_mapgen_id": "fbmw_0",
>     "object": {
>       "mapgensize": [ 6, 6 ],
>       "rows": [
>         "      ",
>         "      ",
>         "      ",
>         "      ",
>         "      ",
>         "      "
>       ],
>       "palettes": [ "fbmw_wood_palette" ]
>     }
>   },
>   {
>     "type": "mapgen",
>     "update_mapgen_id": "faction_base_workshop_0",
>     "method": "json",
>     "object": { "place_nested": [ { "chunks": [ "fbml_0" ], "x": 15, "y": 3 } ] }
>   },
> ```
> 
> that little mapgen really doesn't do anything to the map, but I got errors if I left it out, the survey wants an update_mapgen_id.

Thanks! I'll work on that this week.

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Merge branch 'master' into Modular-Saltworks

Maleclypse committed:
Update modular_saltworks_log.json

Maleclypse committed:
Additional Materials

added other material's for buildings

Maleclypse committed:
added recipes to basecamps

Maleclypse committed:
Create recipe_modular_saltworks_stone.json

Maleclypse committed:
Update modular_saltworks_wad.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_wad.json

Maleclypse commented:
I'm having a JSON validation error " make: *** [data/json/recipes/basecamps/recipe_groups.json] Error 1
Makefile:1034: recipe for target 'data/json/recipes/basecamps/recipe_groups.json' failed" but I'm having difficulty figuring out any differences between my version and the baseline version that would be causing this to fail?

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks_stone.json:
```diff
@@ -0,0 +1,116 @@
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "faction_base_workshop_0",
```
`faction_base_workshop_0` should probably be `faction_base_saltworks_0`

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks_wad.json:
```diff
@@ -0,0 +1,116 @@
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "faction_base_workshop_0",
```
`faction_base_saltworks_0` again

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks_wood.json:
```diff
@@ -0,0 +1,116 @@
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "faction_base_workshop_0",
```
correct to: `faction_base_saltworks_0`

curstwist commented on data/json/mapgen/basecamps/modular_saltworks_wad.json:
```diff
@@ -0,0 +1,116 @@
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "faction_base_workshop_0",
```
`faction_base_saltworks_0`

curstwist commented:
> 
> 
> I'm having a JSON validation error " make: *** [data/json/recipes/basecamps/recipe_groups.json] Error 1
> Makefile:1034: recipe for target 'data/json/recipes/basecamps/recipe_groups.json' failed" but I'm having difficulty figuring out any differences between my version and the baseline version that would be causing this to fail?

you have some references to the workshop that could be the problem.  Also, you only need the survey recipe and its mapgen once, no need to repeat it in every palette.

Maleclypse committed:
updates and corrections

Maleclypse committed:
Update recipe_modular_saltworks_wad.json

Maleclypse committed:
Update recipe_modular_saltworks_wood.json

Maleclypse committed:
Update modular_saltworks_wad.json

Maleclypse committed:
fixes and removing excess pallettes

Maleclypse committed:
fbml to fmbs

Maleclypse committed:
Update recipe_groups.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
Update recipe_modular_saltworks_log.json

Maleclypse committed:
removing { on line 64 of each json

Maleclypse committed:
Update recipe_modular_saltworks_stone.json

Maleclypse committed:
Merge pull request #4 from CleverRaven/master

Catchup to current mainline

Maleclypse committed:
Merge branch 'master' into Modular-Saltworks

Maleclypse committed:
Linting and repair

Maleclypse committed:
Update modular_saltworks_rock.json

Maleclypse committed:
fbms_0 to fbmsw_0

Maleclypse committed:
Update recipe_modular_saltworks_wad.json

Maleclypse committed:
fixed syntax and references

Maleclypse committed:
Update recipe_modular_saltworks_stone.json

Maleclypse committed:
Update overmap_terrain_faction_base.json

Maleclypse committed:
Update modular_saltworks_log.json

Maleclypse committed:
Update faction_buildings.json

Maleclypse committed:
Update recipe_modular_saltworks_wad.json

Maleclypse committed:
Update recipe_modular_saltworks_wood.json

Maleclypse committed:
replace extraneous #

Maleclypse committed:
fbms to fbmsw

Also created saltworks folder for recipes

Maleclypse commented:
All checks clear and testing works. Do I need to close this PR and open a nonWIP version?

anothersimulacrum commented:
Nope, just remove WIP in the title.

Night-Pryanik commented:
Some screenshots?

Maleclypse committed:
add recipe descriptor

Maleclypse commented:
> Some screenshots?


I edited the descriptions.  I also noticed that the brewery upgrade is failing and not calculating the components needed to build so if anyone has any suggestions on how to fix that I'm open to them.  I'll turn this back into WIP until  I get that resolved. When I tested it the first time I just got them all built to first level.  

![Saltworks first level](https://user-images.githubusercontent.com/54345792/67819533-e459cf80-fa8b-11e9-827f-63bb299fa297.png)
![failed](https://user-images.githubusercontent.com/54345792/67819534-e459cf80-fa8b-11e9-8908-dd0598ff2bbc.png)
![recipe failure](https://user-images.githubusercontent.com/54345792/67819535-e459cf80-fa8b-11e9-9cd7-993062ee18af.png)

Maleclypse commented:
@curstwist were you having issues with upgrades to existing buildings and that's why you stopped making basecamp upgrades until it was resolved?  I can't remember.

curstwist commented:
> 
> 
> @curstwist were you having issues with upgrades to existing buildings and that's why you stopped making basecamp upgrades until it was resolved? I can't remember.

I stopped until the rotation issue is fixed (there's a PR up that should fix it) and using the expansion upgrade craft recipies in the "core" building, which isn't working.  Since this is a proper expansion, it shouldn't be either of my issues.  I'll take a look at it.

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_wad.json:
```diff
@@ -0,0 +1,98 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest" } ],
```
typo in blueprint excludes southtwest

Maleclypse replied:
Ok I think I see what I needed to do there now.  I've made changes and will push. I've renamed the upgrades to fbms...1 and 2.

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_metal.json:
```diff
@@ -0,0 +1,92 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest" } ],
```
"blueprint_excludes": [ { "id": "fbmsw_southwest" } ],

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_stone.json:
```diff
@@ -0,0 +1,98 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest" } ],
```
"blueprint_excludes": [ { "id": "fbmsw_souththwest" } ],

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_wood.json:
```diff
@@ -0,0 +1,115 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest" } ],
```
"blueprint_excludes": [ { "id": "fbmsw_souththwest" } ],

Maleclypse committed:
named upgrades appropriately

in blueprint provides requires excludes

curstwist commented:
You can also use "amount": ..either should work.

Maleclypse commented:
I'm not sure what the following means
``src/json.cpp:111: Failed to visit member 'faction_owner' in JsonObject at line 2480:78509:
{
    "type": "mapgen",
    "om_terrain": "faction_base_saltworks_0",
    "method": "json",
    "faction_owner": { "id": "your_followers", "area": [ { "x": 0, "y": 0, "x2": 23, "y2": 23 } ] },
    "weight": 250,``

curstwist commented:
Faction_owner needs to be within the object section. They were all wrong but jbtw recently fixed the others.

AMurkin commented:
Shouldn't `faction_owner` be inside `object`?

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_log.json:
```diff
@@ -0,0 +1,92 @@
+    "construction_blueprint": "fbmsw_brewery2_log_southwest",
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest1" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest2" } ],
```
excludes id is still misspelled. should be identical to requires.

Maleclypse replied:
I followed the same format from the evac shelter recipe? I think?
``    "blueprint_requires": [ { "id": "fbmc_shelter_bedroom1" } ],
    "blueprint_provides": [ { "id": "fbmc_shelter_bedroom2" } ],
    "blueprint_excludes": [ { "id": "fbmc_shelter_bedroom2" } ],``

curstwist replied:
Yeah that's fine, just check the spelling of southwest,  3 are souththwest

Maleclypse replied:
Ah! It's like that thing where as long as the first and last letter are right the brain just fills in the rest.  Thanks clarifying neither I or my wife could find the spelling error.

curstwist replied:
Yeah, it's hard to spot, the eye goes right past it : )

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_metal.json:
```diff
@@ -0,0 +1,92 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest1" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest2" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest2" } ],
```
excludes id is still misspelled. should be identical to requires.

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_wad.json:
```diff
@@ -0,0 +1,98 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest1" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest2" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest2" } ],
```
fbmsw_souththwest2 should be fbmsw_southwest2

curstwist commented on data/json/recipes/basecamps/recipe_modular_saltworks/recipe_modular_saltworks_stone.json:
```diff
@@ -0,0 +1,98 @@
+    "blueprint_name": "southwest brewery2",
+    "blueprint_requires": [ { "id": "fbmsw_southwest1" } ],
+    "blueprint_provides": [ { "id": "fbmsw_southwest2" } ],
+    "blueprint_excludes": [ { "id": "fbmsw_souththwest2" } ],
```
excludes id is still misspelled. should be identical to requires.

Maleclypse committed:
remove excess th's

fixed faction owner

Maleclypse committed:
Update faction_buildings.json

Maleclypse commented:
I'm seeing a bunch of gunmod errors in my C and Travis tests but not anymore errors from files I edited.  Tomorrow I'll see if I can get the brewery upgrades to work and I'll post screenshots of them assuming they do.

Maleclypse committed:
limiting upgrade options

Attempting to make it so it doesn't ask me if I want to replace with other designs

Maleclypse committed:
Lintin new excludes

Maleclypse committed:
changed west to southwest

See if this fixes the ability to upgrade to level 2 brewery

Maleclypse commented:
@curstwist this might just need another set of eyes or maybe I should take a break for the day on it but I've got all the material types building all the way through except for the final brewery upgrade on metal and wad types.  I've also solved my issue where it would offer me to change material types in the base construction screen, I'm not sure if it was the most elegant solution but it seems to have worked by editing the ``blueprint excludes`` line.

Maleclypse committed:
Cursetwist fixes to Saltworks

Maleclypse commented:
Image of completed all materials versions here in Camp Salty.   Top left has a basic brewery, bottom right has upgraded brewery by one level.  All other breweries are level 2.  Left is log, top left is stone, north is metal, top right is wattle and daub, right is wood.  

![Saltworks final](https://user-images.githubusercontent.com/54345792/68098468-e1475080-fe8a-11e9-899f-77c89db8b9a8.png)

Maleclypse committed:
It's the final linting...doo da doot doot
